src/environment/SCRF.cpp

```diff
@@ -161,7 +155,7 @@ void SCRF::accelerateConvergence(mrcpp::ComplexFunction &dfunc, mrcpp::ComplexFu
     dPhi_n.clear();
 }
 
-void SCRF::nestedSCRF(mrcpp::ComplexFunction V_vac) {
+void SCRF::nestedSCRF(mrcpp::ComplexFunction &V_vac, const std::shared_ptr<mrchem::OrbitalVector> &Phi_p) {
```

Comment:
I think this is the appropriate signature here:
1. You're not going to modify the vacuum potential `V_vac`, and
2. You only want to keep the shared pointer alive as long as you use it, but you're not going to change its contents (see screenshot[^1]), and
3. You're not modifying any of the class data, hence the `const` (but this I might be wrong about)
```suggestion
void SCRF::nestedSCRF(const mrcpp::ComplexFunction &V_vac, std::shared_ptr<mrchem::OrbitalVector> Phi_p) const {
```

![2023-10-23_08-40](https://github.com/MRChemSoft/mrchem/assets/3708689/9d72c12b-6ce2-46c0-94ba-f3c9f222f2af)

[^1]: https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer/

Reply:
I am modifying class data in this one. As for the pointer, I think it makes sense to just keep it as copy (it is a pointer afterall).

Reply:
Yes, I see that adding the `const` qualifier to the function is not possible. Still, you can pass `V_vac` by `const` reference. As for passing the `shared_ptr`: it is a pointer, but not all operations on it have the same cost as with a raw pointer. There are different overheads to the different ways of passing it among functions.

Reply:
Ok, I thin it is correct now